Source/Coinbase/CoinbaseApiAuthenticator.cs

```diff
@@ -57,11 +58,15 @@ public void Authenticate(IRestClient client, IRestRequest request)
             var method = request.Method.ToString().ToUpper(CultureInfo.InvariantCulture);
 
             var body = string.Empty;
-
-            var param = request.Parameters.FirstOrDefault(p => p.Type == ParameterType.RequestBody);
-            if (param != null && param?.Value?.ToString() != "null" && !string.IsNullOrWhiteSpace(param?.Value?.ToString()))
-                body = param.Value.ToString();
-
+			if (request.Method != Method.GET) {
```

Comment:
I am using it in production I didn't do a test case on a bit of a crunch but my method that could be converted to a test case pretty easily:

public virtual CoinbaseResponse<Order[]> ListOrders(RequestPagination pagination = null) {  
 190   +			var args = new List<KeyValuePair<string,string>>();  
 191   +			if (pagination != null) {  
 192   +				if (!String.IsNullOrWhiteSpace(pagination.EndingBefore))  
 193   +					args.Add(new KeyValuePair<string, string>("ending_before", pagination.EndingBefore));  
 194   +                if (pagination.Limit != 0)  
 195   +					args.Add(new KeyValuePair<string, string>("limit", pagination.Limit.ToString()));  
 196   +				if (pagination.Order == SortOrder.Asc)  
 197   +					args.Add(new KeyValuePair<string, string>("order", "asc"));  
 198   +				if (!String.IsNullOrWhiteSpace(pagination.StartingAfter))  
 199   +					args.Add(new KeyValuePair<string, string>("starting_after", pagination.StartingAfter));  
 200   +            }  
 201   +			return SendGetRequest<Order[]>("orders",args.ToArray());  
 202   +		}